framework/src/base/Adaptivity.C

```diff
@@ -133,7 +135,7 @@ bool
 Adaptivity::adaptMesh()
 {
   bool meshChanged = false;
-  if (_mesh_refinement_on && (_start_time <= _t && _t < _stop_time))
+  if (_mesh_refinement_on && (_start_time <= _t && _t < _stop_time) && _step % _interval == 0)
```

Comment:
Yes that's fine. This Adaptivity object needs to be cleaned up. It shouldn't be in FEProblem at all.